homeassistant/components/airtouch4/config_flow.py

```diff
@@ -0,0 +1,56 @@
+"""Config flow for AirTouch4."""
+from airtouch4pyapi import AirTouch
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST
+
+from .const import DOMAIN
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+async def _validate_connection(hass: core.HomeAssistant, host):
+    airtouch = AirTouch(host)
+    await airtouch.UpdateInfo()
+
+    if hasattr(airtouch, "error"):
+        if isinstance(airtouch.error, Exception):
```

Comment:
If there's an error the library should raise it. Please refactor this. Please have all errors be raised exceptions. The library user should not have to distinguish them like this.

Reply:
For the purposes of a HA PR can we just assume that this is how the external library works?

Reply:
No, this is a really bad pattern, please fix it.

Reply:
Ok, this might take a while... the reason why it is like this is because while testing we had issues with the library raising errors as errors and HA hitting an unrecoverable state for something that would be fixed up on the next loop around (eg. a temporary network drop) - changing this will require extensive re-testing

Reply:
Sounds like a good idea to properly address this before we release the integration.

Reply:
Don't know how to address everyones routers!

Reply:
Just because an exception is raised doesn't mean it will leak and stop a process. If the library raises a known exception the library user can catch that and act appropriately.

Reply:
ah I understand. Was modelling alot from the way this integration works but maybe I've taken it in the wrong direction - https://github.com/home-assistant/core/blob/dev/homeassistant/components/coolmaster/config_flow.py#L18 Perhaps better would be to avoid the use of exceptions entirely, it could probably just be a flag instead.

Reply:
It depends on what we want to communicate to the library user. If it's some sort of error or missing information state that requires the user to change flow from a "normal" flow, an exception is probably appropriate. If it's an option for an optional feature, it could eg be a boolean attribute. It's hard to give good advice without seeing the whole picture, so you should decide in the end. I can only generally identify good and bad patterns and when I read the PR.